scripts/lint-style.py

```diff
@@ -149,6 +152,45 @@ def long_lines_check(lines, path):
             errors += [(ERR_LIN, line_nr, path)]
     return errors
 
+def indent_check(lines, path):
+    errors = []
+    indent_lvl = 0
+    in_prf = 0  # counter for nested proof blocks
+    check_rest_of_block = True  # we only check uncomplicated syntax
+    ended_with_comma = False  # track whether the previous line ends with a comma
+    inside_special = 0  # track whether we are inside ⟨⟩ or []
+    for line_nr, line in enumerate(lines, 1):
+        lstr = line.lstrip(' ')  # strip spaces from beginning of line
+        if in_prf > 0 and check_rest_of_block and ended_with_comma and not inside_special:
+            if lstr[0] == '{' and len(line) - len(lstr) != indent_lvl:
+                errors += [(WRN_IND, line_nr, path)]
+        ended_with_comma = (line.endswith(",\n"))
+        inside_special += line.count('⟨') + line.count('[') - line.count('⟩') - line.count(']')
+        if line[0] != ' ':
+            # reset the state
+            indent_lvl = 0
+            in_prf = 0
+            check_rest_of_block = True
+            ended_with_comma = False
+            inside_special = 0
+        if "match" in line or "calc" in line:
+            check_rest_of_block = False
+        if "begin" in line:
+            if line.find("begin") > 0 and in_prf == 0:
+                # complicate proof block syntax
+                check_rest_of_block = False
+            if lstr.find("begin") > 0:
+                # complicate proof block syntax
+                check_rest_of_block = False
+            indent_lvl += 2
+            in_prf += 1
+        if "end" in line:
+            indent_lvl -= 2
+            in_prf -= 1
+        indent_lvl += 2 * line.count('{') # potential innocent(?) clash with set-builder notation
+        indent_lvl -= 2 * line.count('}') # there can be multiple closing braces on one line
```

Comment:
What is the motivation for checking `[`/`]` indent level before the "reset" logic and `{`/`}` indent level afterwards?

Reply:
`[`/`]` is checking the nesting level, and if we're nested inside such a special block we don't want to do anything.
Moving these `indent_lvl` lines for `{`/`}` would break the indent checking code in the lines above.